nodebuilder/p2p/bitswap.go

```diff
@@ -36,10 +30,11 @@ func dataExchange(tp node.Type, params bitSwapParams) exchange.SessionExchange {
 	switch tp {
 	case node.Full, node.Bridge:
 		bs := bitswap.New(params.Ctx, net, params.Bs)
+		net.Start(bs.Client, bs.Server)
```

Comment:
Did you try extracting bitswap construction to share pkg?  
(we can do that later as well)